knative-operator/pkg/controller/knativekafka/knativekafka_controller_test.go

```diff
@@ -0,0 +1,500 @@
+package knativekafka
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	mf "github.com/manifestival/manifestival"
+	v1alpha1 "github.com/openshift-knative/serverless-operator/knative-operator/pkg/apis/operator/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	admissiontypes "sigs.k8s.io/controller-runtime/pkg/webhook/admission/types"
+)
+
+var (
+	defaultRequest = reconcile.Request{
+		NamespacedName: types.NamespacedName{Namespace: "knative-eventing", Name: "knative-kafka"},
+	}
+)
+
+type fakeManager struct {
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+func (*fakeManager) Add(manager.Runnable) error {
+	return nil
+}
+
+func (*fakeManager) SetFields(interface{}) error {
+	return nil
+}
+
+func (*fakeManager) Start(<-chan struct{}) error {
+	return nil
+}
+
+func (*fakeManager) GetConfig() *rest.Config {
+	return nil
+}
+
+func (f *fakeManager) GetScheme() *runtime.Scheme {
+	return f.scheme
+}
+
+func (*fakeManager) GetAdmissionDecoder() admissiontypes.Decoder {
+	return nil
+}
+
+func (f *fakeManager) GetClient() client.Client {
+	return f.client
+}
+
+func (*fakeManager) GetFieldIndexer() client.FieldIndexer {
+	return nil
+}
+
+func (*fakeManager) GetCache() cache.Cache {
+	return nil
+}
+
+func (*fakeManager) GetRecorder(name string) record.EventRecorder {
+	return nil
+}
+
+func (*fakeManager) GetRESTMapper() meta.RESTMapper {
+	return nil
+}
+
+var _ manager.Manager = &fakeManager{}
+
+func init() {
+	os.Setenv("OPERATOR_NAME", "TEST_OPERATOR")
+	os.Setenv("KAFKACHANNEL_MANIFEST_PATH", "testdata/kafkachannel-latest.yaml")
+	os.Setenv("KAFKASOURCE_MANIFEST_PATH", "testdata/kafkasource-latest.yaml")
+}
+
+func TestKnativeKafkaReconcile(t *testing.T) {
+	logf.SetLogger(logf.ZapLogger(true))
+
+	tests := []struct {
+		name         string
+		instance     *v1alpha1.KnativeKafka
+		exists       []types.NamespacedName
+		doesNotExist []types.NamespacedName
+	}{
+		{
```

Comment:
nit: You can collapse the tests a bit to get rid of a level of indentation, you can do:

```golang
tests := []struct{
  ...
}{{
  // your first test
}, {
  // your second test
}}
```

nothing blocking obviously, I personally prefer writing (and more importantly reading) that way though. Can be applied to all tests.